usr/src/uts/common/fs/zfs/sys/zio.h

```diff
@@ -209,12 +209,13 @@ enum zio_flag {
 	ZIO_FLAG_CANFAIL)
 
 enum zio_child {
-	ZIO_CHILD_VDEV = 0,
-	ZIO_CHILD_GANG,
-	ZIO_CHILD_DDT,
-	ZIO_CHILD_LOGICAL,
-	ZIO_CHILD_TYPES
+	ZIO_CHILD_VDEV		= 1 << 0,
+	ZIO_CHILD_GANG		= 1 << 1,
+	ZIO_CHILD_DDT		= 1 << 2,
+	ZIO_CHILD_LOGICAL	= 1 << 3
```

Comment:
I do not have any argument against this change, but have you considered keeping this enumeration as is, but instead changing only `zio_wait_for_children` to expect the bit-mask?
E.g. something like
```
#define ZIO_CHILD_BIT(c) (1 << (c))
```
and then
```
if (zio_wait_for_children(zio,
    ZIO_CHILD_BIT(ZIO_CHILD_GANG) | ZIO_CHILD_BIT(ZIO_CHILD_LOGICAL),
    ZIO_WAIT_READY)) {
...
}
```

Looks a bit too verbose, but has a benefit of affecting only `zio_wait_for_children` calls which need to be modified in either case.

Reply:
Yeah, I looked at this option but didn't like how it looked specifically for this reason. I've looked at using variable arguments too but it has drawback too. I'm going to consider this option again but using a variable to hold the bits we plan to pass into `zio_wait_for_children`.